### PR TITLE
Fix app_management_test at HEAD

### PR DESCRIPTION
**Pull request by drigz**
See #447. Tested with:

```
bazel test //src/app_charts/base:app_management_test
USE_BAZEL_VERSION=b084956e23a6e50e35fe8886d0f3f7e05f718155 bazel test //src/app_charts/base:app_management_test
```

### CODE REVIEW

File: src/app_charts/base/app_management_test.sh [hunk 1]
```diff
@@ -14,7 +14,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-HELM="${TEST_SRCDIR}/_main/external/_main~non_module_deps~kubernetes_helm/helm template"
+if [[ -x "${TEST_SRCDIR}/_main/external/_main~non_module_deps~kubernetes_helm/helm" ]] ; then
+  HELM="${TEST_SRCDIR}/_main/external/_main~non_module_deps~kubernetes_helm/helm template"
+elif [[ -x "${TEST_SRCDIR}/+non_module_deps+kubernetes_helm/helm" ]] ; then
+  HELM="${TEST_SRCDIR}/+non_module_deps+kubernetes_helm/helm template"
+else
```
Comment (ensonic) [inline src/app_charts/base/app_management_test.sh:21]: can we do the same in deploy.sh? Any idea why my find approach failed? Btw. my find approach worked locally, but failed when I manually triggered the integration tests action workflow.

Comment (ensonic) [reply src/app_charts/base/app_management_test.sh:21]: Maybe less copy'n'paste if we write:
```
HELM="${TEST_SRCDIR}/_main/external/_main~non_module_deps~kubernetes_helm/helm"
if [[ ! -x "${HELM}" ]]; then
  HELM="${TEST_SRCDIR}/+non_module_deps+kubernetes_helm/helm template"
fi
if [[ ! -x "${HELM}" ]]; then
  echo >&2 "Failed to locate helm in ${TEST_SRCDIR}."
  exit 1
fi
```

Comment (drigz) [reply src/app_charts/base/app_management_test.sh:21]: Done. I copied it to deploy.sh, but I was too lazy to reimplement the logic in Go, so I just left a comment that hopefully someone will find when they update the Bazel pin.

**Review by ensonic — APPROVED** [review]

**Comment by ensonic** [comment]
Sadly it fails on our CI, same as my change failed:
https://github.com/googlecloudrobotics/core/actions/runs/11034197473/job/30647145593

**Comment by drigz** [comment]
Ah, thanks for the link. I think it's because the build hasn't happened at that point of deploy.sh. We should probably just keep it hardcoded there and update it when the pin changes.

**Comment by ensonic** [comment]
Ahh, that's a bit silly. +1 for then just adding another TODO there. Would be much better to just set this further own in the script.

**Pull request merged**